tests/helpers/main.cu

```diff
@@ -1,27 +1,60 @@
 #include <cuda_runtime.h>
 #include <cstdio>
+#include <map>
 
 #include "flamegpu/gpu/CUDASimulation.h"
 #include "gtest/gtest.h"
 #include "helpers/device_initialisation.h"
+#include "flamegpu/util/Environment.h"
+#include "flamegpu/io/Telemetry.h"
 
 
 GTEST_API_ int main(int argc, char **argv) {
-  // Time the cuda agent model initialisation, to check it creates the context.
-  flamegpu::tests::timeCUDASimulationContextCreationTest();
-  // Run the main google test body
-  printf("Running main() from %s\n", __FILE__);
-  testing::InitGoogleTest(&argc, argv);
-  auto rtn = RUN_ALL_TESTS();
-  // Reset all cuda devices for memcheck / profiling purposes.
-  int devices = 0;
-  gpuErrchk(cudaGetDeviceCount(&devices));
-  if (devices > 0) {
-    for (int device = 0; device < devices; ++device) {
-      gpuErrchk(cudaSetDevice(device));
-      gpuErrchk(cudaDeviceReset());
+    // Telemetry
+    const bool share_usage = flamegpu::io::Telemetry::globalTelemetryEnabled();
+    flamegpu::util::setEnvironmentVariable("FLAMEGPU_SHARE_USAGE_STATISTICS", "False");    // Disable any global telemetry to prevent telemetry for every simulation in test suite
```

Comment:
Set a namesapce scoped variable here isntead of using an env var.

Reply:
Why? A global variable is already in place to control this behaviour.

Reply:
`setEnvironmentVariable` is setting a temporary system environment variable here though, there's no reason to bother with the faff of cross platform environment variable settings (which may already exist due to users being able to manually set them, not sure how that interaction works though). Which global are you referring to? We don't want any symbols outside our namespaces though.